packages/opentelemetry-scope-zone-peer-dep/src/ZoneScopeManager.ts

```diff
@@ -0,0 +1,511 @@
+      enumerable: false,
+      configurable: true,
+      writable: false,
+      value: target.length,
+    });
+    return (contextWrapper as unknown) as T;
+  }
+
+  /**
+   * @param obj target object on which the listeners will be patched
+   * @param scope
+   */
+  private _bindListener<T>(obj: T, scope?: unknown): T {
+    const target = (obj as unknown) as TargetWithEvents;
+    if (target.__ot_listeners !== undefined) {
+      return obj;
+    }
+    target.__ot_listeners = {};
+
+    if (typeof target.addEventListener === 'function') {
+      target.addEventListener = this._patchAddEventListener(
+        target,
+        target.addEventListener,
+        scope
+      );
+    }
+
+    if (typeof target.removeEventListener === 'function') {
+      target.removeEventListener = this._patchRemoveEventListener(
+        target,
+        target.removeEventListener,
+        scope
+      );
+    }
+
+    return obj;
+  }
+
+  /**
+   * Removes references to the targets that were bind to certain scope / span
+   * @param scope
+   */
+  private _cleanEventTargetsFromScope(scope: any, target?: TargetWithEvents) {
+    // console.log('_cleanEventTargetsFromScope1', this._scopesWithEventTargets.length, scope && scope.name);
+    for (let i = this._scopesWithEventTargets.length - 1; i >= 0; i--) {
+      const scopeWithEventTargets = this._scopesWithEventTargets[i];
+      if (scopeWithEventTargets.scope === scope) {
+        if (
+          typeof target !== 'undefined' &&
+          scopeWithEventTargets.target !== target
+        ) {
+          continue;
+        }
+
+        if (scopeWithEventTargets.target) {
+          try {
+            scopeWithEventTargets.target.addEventListener =
+              scopeWithEventTargets.eventTargets.addEventListener;
+
+            scopeWithEventTargets.target.removeEventListener =
+              scopeWithEventTargets.eventTargets.removeEventListener;
+
+            delete scopeWithEventTargets.target.__ot_listeners;
+          } catch (e) {
+            console.log(e);
+          }
+        }
+
+        this._scopesWithEventTargets.splice(i, 1);
+
+        if (
+          typeof target !== 'undefined' &&
+          scopeWithEventTargets.target === target
+        ) {
+          break;
+        }
+      }
+    }
+    // console.log('_cleanEventTargetsFromScope2', this._scopesWithEventTargets.length);
+  }
+
+  /**
+   * Removes all the associated scopes with zone after the zone has no more tasks
+   * @param zoneName
+   */
+  private _cleanScopesFromZone(zoneName: string) {
+    const scopes = this._scopes[zoneName];
+    for (let i = scopes.length - 1; i >= 0; i--) {
+      this._cleanEventTargetsFromScope(scopes[i]);
+      scopes.splice(i, 1);
+    }
+    delete this._scopes[zoneName];
+  }
+
+  /**
+   * Creates a new unique zone name
+   */
+  private _createZoneName() {
+    this._zoneCounter++;
+    const random = Math.random();
+    const zoneName = `${this._zoneCounter}-${random}`;
```

Comment:
Could you just return `${this._zoneCounter}-${random}` and so remove the `zoneName` variable?